pkg/diskmaker/discovery/discovery.go

```diff
@@ -0,0 +1,262 @@
+					klog.Errorf("failed to discover devices triggered from udev event. %v", err)
+				}
+			} else {
+				klog.Warningf("disabling udev monitoring")
+				udevEvents = nil
+			}
+		}
+	}
+}
+
+// discoverDevices identifies the list of usable disks on the current node
+func (discovery *DeviceDiscovery) discoverDevices() error {
+	// List all the valid block devices on the node
+	validDevices, err := getValidBlockDevices()
+	if err != nil {
+		message := "failed to discover devices"
+		e := diskmaker.NewEvent(diskmaker.ErrorListingBlockDevices, fmt.Sprintf("%s. Error: %+v", message, err), "")
+		discovery.eventSync.Report(e, discovery.localVolumeDiscovery)
+		return errors.Wrapf(err, message)
+	}
+
+	klog.Infof("valid block devices: %+v", validDevices)
+
+	discoveredDisks := getDiscoverdDevices(validDevices)
+	klog.Infof("discovered devices: %+v", discoveredDisks)
+
+	// Update discovered devices in the  LocalVolumeDiscoveryResult resource
+	if !reflect.DeepEqual(discovery.disks, discoveredDisks) {
+		klog.Info("device list updated. Updating LocalVolumeDiscoveryResult status...")
+		discovery.disks = discoveredDisks
+		err = discovery.updateStatus()
+		if err != nil {
+			message := "failed to update LocalVolumeDiscoveryResult status"
+			e := diskmaker.NewEvent(diskmaker.ErrorUpdatingDiscoveryResultObject, fmt.Sprintf("%s. Error: %+v", message, err), "")
+			discovery.eventSync.Report(e, discovery.localVolumeDiscovery)
+			return errors.Wrapf(err, message)
+		}
+		message := "successfully updated discovered device details in the LocalVolumeDiscoveryResult resource"
+		e := diskmaker.NewSuccessEvent(diskmaker.UpdatedDiscoveredDeviceList, message, "")
+		discovery.eventSync.Report(e, discovery.localVolumeDiscovery)
+	}
+
+	return nil
+}
+
+// getValidBlockDevices fetchs all the block devices sutitable for discovery
+func getValidBlockDevices() ([]internal.BlockDevice, error) {
+	blockDevices, badRows, err := internal.ListBlockDevices()
+	if err != nil {
+
+		return blockDevices, errors.Wrapf(err, "failed to list all the block devices in the node.")
+	} else if len(badRows) > 0 {
+		klog.Warningf("failed to parse all the lsblk rows. Bad rows: %+v", badRows)
+	}
+
+	// Get valid list of devices
+	validDevices := make([]internal.BlockDevice, 0)
+	for _, blockDevice := range blockDevices {
+		if ignoreDevices(blockDevice) {
+			continue
+		}
+		validDevices = append(validDevices, blockDevice)
+	}
+
+	return validDevices, nil
+}
+
+// getDiscoverdDevices creates v1alpha1.DiscoveredDevice from internal.BlockDevices
+func getDiscoverdDevices(blockDevices []internal.BlockDevice) []v1alpha1.DiscoveredDevice {
+	discoveredDevices := make([]v1alpha1.DiscoveredDevice, 0)
+	for _, blockDevice := range blockDevices {
+		deviceID, err := blockDevice.GetPathByID()
+		if err != nil {
+			klog.Warningf("failed to get persisent ID for the device %q. Error %v", blockDevice.Name, err)
+			deviceID = ""
+		}
+		size, err := resource.ParseQuantity(blockDevice.Size)
```

Comment:
One of the motivations of defining size field as `resource.Quantity` is, so as it can seen/used as "10Gi" or "100GiB" in api objects. Using these values as such (as integers) defeats the purpose.

Reply:
@gnufied Can we have this change as part of bug fix. Because this will cause breaking changes in the UI and UI code is already merged.

Reply:
wait, what breaking changes in the UI? I do not understand

Reply:
what I am trying to say if we are going to report size as integer bytes, then there was no point in picking `resource.Quantity` type for the size field.

Reply:
UI calculates discovered capacity. It does not handle every available format of resource.Quantity yet.